lib/MiLight/FUT089PacketFormatter.cpp

```diff
@@ -111,22 +111,30 @@ BulbId FUT089PacketFormatter::parsePacket(const uint8_t *packet, JsonObject resu
 
   uint8_t command = (packetCopy[V2_COMMAND_INDEX] & 0x7F);
   uint8_t arg = packetCopy[V2_ARGUMENT_INDEX];
+  bool long_press = (packetCopy[V2_COMMAND_INDEX] & 0x80) == 0x80;
 
   if (command == FUT089_ON) {
-    if ((packetCopy[V2_COMMAND_INDEX] & 0x80) == 0x80) {
-      result[GroupStateFieldNames::COMMAND] = MiLightCommandNames::NIGHT_MODE;
-    } else if (arg == FUT089_MODE_SPEED_DOWN) {
+    if (arg == FUT089_MODE_SPEED_DOWN) {
       result[GroupStateFieldNames::COMMAND] = MiLightCommandNames::MODE_SPEED_DOWN;
+      result[GroupStateFieldNames::LONG_PRESS] = long_press;
     } else if (arg == FUT089_MODE_SPEED_UP) {
       result[GroupStateFieldNames::COMMAND] = MiLightCommandNames::MODE_SPEED_UP;
+      result[GroupStateFieldNames::LONG_PRESS] = long_press;
     } else if (arg == FUT089_WHITE_MODE) {
       result[GroupStateFieldNames::COMMAND] = MiLightCommandNames::SET_WHITE;
+      result[GroupStateFieldNames::LONG_PRESS] = long_press;
     } else if (arg <= 8) { // Group is not reliably encoded in group byte. Extract from arg byte
       result[GroupStateFieldNames::STATE] = "ON";
       bulbId.groupId = arg;
+      result[GroupStateFieldNames::LONG_PRESS] = long_press;
     } else if (arg >= 9 && arg <= 17) {
       result[GroupStateFieldNames::STATE] = "OFF";
       bulbId.groupId = arg-9;
+      result[GroupStateFieldNames::LONG_PRESS] = long_press;
+      if (long_press) {
+        result[GroupStateFieldNames::STATE] = "ON";
```

Comment:
This changes the behavior of the lights. Now when I keep pressing the off button, it will show one "OFF" state and then "ON" with night_mode. Before it was continuous "OFF" with night mode. I am not sure which way is better, but to me it is confusing that I am pressing off but the command shows as on.